sycl/include/CL/sycl/handler.hpp

```diff
@@ -918,10 +918,13 @@ class __SYCL_EXPORT handler {
 
   template <int Dims, typename LambdaArgType> struct TransformUserItemType {
     using type = typename std::conditional<
-        std::is_convertible<nd_item<Dims>, LambdaArgType>::value, nd_item<Dims>,
+        std::is_convertible<nd_item<Dims>, LambdaArgType>::value &&
+            !std::is_same<id<Dims>, LambdaArgType>::value,
+        nd_item<Dims>,
         typename std::conditional<
-            std::is_convertible<item<Dims>, LambdaArgType>::value, item<Dims>,
-            LambdaArgType>::type>::type;
+            std::is_convertible<item<Dims>, LambdaArgType>::value &&
+                !std::is_same<id<Dims>, LambdaArgType>::value,
```

Comment:
can you rework it to check id only once?

Reply:
Done